### PR TITLE
FSDP (native) support for LightningLite

### LINKED ISSUE #14898: Support FSDP in Lite
### Motivation

Native PyTorch FSDP is a powerful alternative to regular DDP. 

### Pitch

Add the strategy to Lite. Build it using the existing implementation in Trainer. 

______________________________________________________________________

#### If you enjoy Lightning, check out our other projects! ⚡

- [**Metrics**](https://github.com/Lightning-AI/metrics): Machine learning metrics for distributed, scalable PyTorch applications.

- [**Lite**](https://pytorch-lightning.readthedocs.io/en/latest/starter/lightning_lite.html): enables pure PyTorch users to scale their existing code on any kind of device while retaining full control over their own loops and optimization logic.

- [**Flash**](https://github.com/Lightning-AI/lightning-flash): The fastest way to get a Lightning baseline! A collection of tasks for fast prototyping, baselining, fine-tuning, and solving problems with deep learning.

- [**Bolts**](https://github.com/Lightning-AI/lightning-bolts): Pretrained SOTA Deep Learning models, callbacks, and more for research and production with PyTorch Lightning and PyTorch.

- [**Lightning Transformers**](https://github.com/Lightning-AI/lightning-transformers): Flexible interface for high-performance research using SOTA Transformers leveraging PyTorch Lightning, Transformers, and Hydra.

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

Adds support for `strategy="fsdp"` in LightningLite.

Fixes #14898

TODOs:
- [x] no sync after #14966  (see also [this](https://pytorch.org/docs/stable/fsdp.html#torch.distributed.fsdp.FullyShardedDataParallel.no_sync) and [this](https://github.com/pytorch/xla/blob/44104e7e8dd703b844ec16dd053520e24965e24d/torch_xla/distributed/fsdp/xla_fully_sharded_data_parallel.py#L667))

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
I made sure I had fun coding 🙃

cc @borda @carmocca @justusschock @awaelchli

### CODE REVIEW

File: tests/tests_pytorch/strategies/test_ddp_fully_sharded_native.py [hunk 1]
```diff
@@ -148,6 +148,7 @@ def custom_auto_wrap_policy(
     return unwrapped_params >= 2
 
 
+# lite: adopted
```
Comment (awaelchli) [inline tests/tests_pytorch/strategies/test_ddp_fully_sharded_native.py:151]: Show reviewers that we adopted all tests that are in PL. 
TODO: remove before merge

**Review by justusschock — APPROVED** [review]

**Review by Borda — APPROVED** [review]

**Review by tchaton — APPROVED** [review]
LGTM !

**Review by carmocca — APPROVED** [review]
